### PR TITLE
Remove unused ConstantTypeList, and TypeList, + fixes a few inconsist…

### PR DESCRIPTION
**Pull request by erickvermot**
…encies regarding list items sizes.

relates to https://github.com/kytos/python-openflow/issues/418

### CODE REVIEW

File: pyof/foundation/basic_types.py [hunk 1]
```diff
@@ -478,33 +478,22 @@ class attribute of the struct.
...
-                type.
+            pyof_class (:obj:`type`): Class of the items to be stored.
+            items (iterable, ``pyof_class``): Items to be stored.
         """
-        for item in items:
-            self.append(item)
+        self._pyof_class = pyof_class
+        if items is None:
+            items = []
+        super().__init__(items)
```
Comment (erickvermot) [inline pyof/foundation/basic_types.py]: 1: This is the same behavior of any of the other classes. UBInt8('a') for example is perfectly valid, and it only complains when trying to pack. The list is just keeping the same behavior.

2: This allows FixedTypeList(UBInt8, [1,2,3]) for example. Checking the class does not.

File: pyof/foundation/basic_types.py [hunk 2]
```diff
@@ -517,41 +506,18 @@ def pack(self, value=None):
...
             value = self
-        else:
-            container = type(self)(items=None)
-            container.extend(value)
-            value = container
 
         bin_message = b''
         try:
             for item in value:
+                if not isinstance(item, self._pyof_class):
```
Comment (erickvermot) [inline pyof/foundation/basic_types.py]: same as previous comment:


1: This is the same behavior of any of the other classes. UBInt8('a') for example is perfectly valid, and it only complains when trying to pack. The list is just keeping the same behavior.

2: This allows FixedTypeList(UBInt8, [1,2,3]) for example. Checking the class does not.

File: pyof/foundation/basic_types.py [hunk 3]
```diff
@@ -567,142 +533,24 @@ def get_size(self, value=None):
...
-        """The constructor parameters follows.
-
-        Args:
-            pyof_class (:obj:`type`): Class of the items to be stored.
-            items (iterable, ``pyof_class``): Items to be stored.
-        """
-        self._pyof_class = pyof_class
-        super().__init__(items)
-
-    def append(self, item):
```
Comment (erickvermot) [inline pyof/foundation/basic_types.py]: Same as previous comment...

File: pyof/foundation/basic_types.py [hunk 4]
```diff
@@ -567,142 +533,24 @@ def get_size(self, value=None):
...
-        """
-        if isinstance(item, list):
-            self.extend(item)
-        elif issubclass(item.__class__, self._pyof_class):
-            list.append(self, item)
-        else:
-            raise exceptions.WrongListItemType(item.__class__.__name__,
-                                               self._pyof_class.__name__)
-
-    def insert(self, index, item):
```
Comment (erickvermot) [inline pyof/foundation/basic_types.py]: Same as previous comment...

File: pyof/foundation/basic_types.py [hunk 5]
```diff
@@ -9,7 +9,7 @@
 
 # Third-party imports
 
-__all__ = ('BinaryData', 'Char', 'ConstantTypeList', 'FixedTypeList',
+__all__ = ('BinaryData', 'Char', 'FixedTypeList',
```
Comment (erickvermot) [inline pyof/foundation/basic_types.py]: ok

**Comment by beraldoleal** [comment]
Since that the requested changes were not made, I'm closing this.

**Pull request closed without merging**